lib/tbot/workloadidentity/workloadattest/container/kubernetes.go

```diff
@@ -0,0 +1,87 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package container
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// Kubernetes discovers Kubernetes containers and pods.
+var Kubernetes Engine = k8s{}
+
+type k8s struct{}
+
+// Note: this is a fairly naive implementation, we may need to make further
+// improvements to account for other distributions of Kubernetes.
+//
+// There's a collection of real world mountfiles in testdata/mountfile.
+func (k8s) parseCgroupMount(source string) (*Info, error) {
+	matches := k8sContainerIDRegex.FindStringSubmatch(source)
+	if len(matches) != 2 {
+		return nil, trace.BadParameter(
+			"expected 2 matches searching for container ID but found %d",
+			len(matches),
+		)
+	}
+
+	containerID := matches[1]
+	if containerID == "" {
+		return nil, trace.BadParameter(
+			"source does not contain container ID",
+		)
+	}
+
+	matches = k8sPodIDRegex.FindStringSubmatch(source)
+	if len(matches) != 2 {
+		return nil, trace.BadParameter(
+			"expected 2 matches searching for pod ID but found %d",
+			len(matches),
+		)
+	}
+	podID := matches[1]
+	if podID == "" {
+		return nil, trace.BadParameter(
+			"source does not contain pod ID",
+		)
+	}
+
+	// When using the `systemd` cgroup driver, the dashes are replaced with
+	// underscores. So let's correct that.
+	podID = strings.ReplaceAll(podID, "_", "-")
+
+	return &Info{
+		ID:    containerID,
+		PodID: podID,
+	}, nil
+}
+
+var (
```

Comment:
Nit: Maybe we ought to move these up to the top.

Reply:
Sure, no strong opinions here - I just default to putting exported members higher up in the file.